pytrainsim/OCPSim/NetworkParser.py

```
@@ -1,34 +1,225 @@
+from math import ceil, sin, cos, sqrt, atan2, radians
 
 
-def network_from_json(json_data: str) -> "Network[Track]":
-    data = json.loads(json_data)
-    network = Network[Track]()
-    ocps = []
-    for ocp_data in data["ocps"]:
-        ocp = OCP[Track](ocp_data["db640_code"])
-        if "latitude" in ocp_data and "longitude" in ocp_data:
-            ocp.geo = GeoPoint(ocp_data["latitude"], ocp_data["longitude"])
-        ocps.append(ocp)
-    network.add_ocps(ocps)
+import xml.etree.ElementTree as ET
 
-    for track_data in data["tracks"]:
-        start = network.get_ocp(track_data["start"])
-        end = network.get_ocp(track_data["end"])
 
-        if start is None or end is None:
+@dataclass
+class TrackData:
+    ocp1: str
+    ocp2: str
+    length: int = 10
+    max_speed: float = 100 / 3.6
+    capacity_up: int = 0
+    capacity_down: int = 0
+    capacity_none: int = 0
+
+    def __post_init__(self):
+        if self.ocp1 > self.ocp2:
+            self.ocp1, self.ocp2 = self.ocp2, self.ocp1
+            self.capacity_down, self.capacity_up = self.capacity_up, self.capacity_down
+
+    def add_capacity(self, other: TrackData) -> None:
+        if self.ocp1 != other.ocp1 or self.ocp2 != other.ocp2:
+            raise ValueError("Cannot add capacities for different tracks")
+
+        self.capacity_up += other.capacity_up
+        self.capacity_down += other.capacity_down
+        self.capacity_none += other.capacity_none
+
+    def key(self) -> str:
+        return f"{self.ocp1}-{self.ocp2}"
+
+    @staticmethod
+    def _get_max_speed(track_element: ET.Element, namespaces: Dict[str, str]) -> float:
+        max_speed_element: Optional[ET.Element] = track_element.find(
+            "railml:trackElements/railml:speedChanges/railml:speedChange", namespaces
+        )
+        return (
+            float(max_speed_element.attrib["vMax"]) / 3.6
+            if max_speed_element is not None
```

raise error instead of default